cli/sync_service/object.go

```diff
@@ -5,104 +5,170 @@ import (
 	Destinations []common.DestinationsStatus `json:"destinations,omitempty"`
+	ObjectStatus string                      `json:"objectStatus,omitempty"`
 }
 
-// Display the object metadata for a given object in the MMS.
-func ObjectList(org string, userPw string, objType string, objId string, details bool) {
+// Display the object metadata for given flags in the MMS.
+func ObjectList(org string, userPw string, objType string, objId string, destPolicy string, dpService string, dpPropertyName string, dpUpdateTimeSince string, destType string, destId string, withData string, expirationTimeBefore string, long bool, details bool) {
 	// get message printer
 	msgPrinter := i18n.GetMessagePrinter()
 
 	if userPw == "" {
 		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("must specify exchange credentials to access the model management service."))
 	}
 
-	if details && objId == "" {
-		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("must specify object ID when requesting object details."))
-	}
-
 	// Set the API key env var if that's what we're using.
 	cliutils.SetWhetherUsingApiKey(userPw)
 
-	// If object ID is omitted, query all objects of the given type.
-	if !details {
-		objectList := new(exchange.ObjectDestinationPolicies)
-		urlPath := path.Join("api/v1/objects/", org, objType, "?all_objects=true")
+	var objectsMeta []common.MetaData
+
+	// validate params
+	if strings.ToLower(destPolicy) != "true" && strings.ToLower(destPolicy) != "false" {
+		destPolicy = ""
+	} else {
+		destPolicy = strings.ToLower(destPolicy)
+	}
+
+	noData := ""
+	if strings.ToLower(withData) == "true" || strings.ToLower(withData) == "false" {
+		withDataBool, _ := strconv.ParseBool(strings.ToLower(withData))
+		noData = strconv.FormatBool(!withDataBool)
+	}
 
-		// Call the MMS service over HTTP to get the basic object metadata.
-		httpCode := cliutils.ExchangeGet("Model Management Service", cliutils.GetMMSUrl(), urlPath, cliutils.OrgAndCreds(org, userPw), []int{200, 404}, objectList)
-		if httpCode == 404 {
-			cliutils.Fatal(cliutils.NOT_FOUND, msgPrinter.Sprintf("no objects type '%s' found in org %s", objType, org))
+	if dpService != "" || dpPropertyName != "" || dpUpdateTimeSince != "" {
+		if destPolicy == "false" {
+			cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("must specify destinationPolicy to true or set to empty when filter by --service, --property, or --updateTime"))
 		}
 
-		output := ""
-		if objId == "" {
-			output = cliutils.MarshalIndent(objectList, "mms object list")
+		if !dpServiceIsValid(dpService) {
+			cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("service should be in format service-org/service-name"))
+		}
+
+		timeValidated, convTimeStamp := timeIsValid(dpUpdateTimeSince)
+		if !timeValidated {
+			cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("updateTime should be in RC3339 format: yyyy-MM-dd'T'HH:mm:ssZ, or yyyy-MM-dd"))
```

Comment:
Please remove the single quotes around the T. I think it might be more clear to the user this way.